lib/recurly/validate.js

```diff
@@ -15,8 +15,49 @@ const debug = require('debug')('recurly:validate');
 const TYPES = [
   {
     type: 'discover',
-    pattern: /^(3[0689]|6011|621094|62[24568]|64[4-9]|65|81[0-7])/,
-    lengths: [14, 15, 16, 17, 18, 19]
+    pattern: {
+      test: (n) => {
+        let firstEight = parseInt(n.substr(0, 8), 10);
+        let accepted_ranges = [
+          [60110000,60110399],
+          [60110500,60110999],
+          [60112000,60114999],
+          [60117400,60117499],
+          [60117700,60117999],
+          [60118600,60119999],
+          [64400000,65059999],
+          [65060100,65060999],
+          [65061100,65999999]
+        ];
+
+        if (!(/^6[045]/.test(firstEight))) return false;
+        return accepted_ranges.indexOf(([minimum, maximum]) => firstEight >= minimum && firstEight <= maximum) !== undefined;
```

Comment:
[`Array#indexOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) doesn't accept a function like this, and the return value will always be `-1` here, thus this comparison will always return true.

Reply:
Refactored that one for both discover and union_pay.